File: DevExpress-Demo/CS/GaugesDemo.Wpf/ViewModels/EqualizerViewModel.cs
using System;
using System.Windows.Threading;

namespace GaugesDemo {
    public class EqualizerViewModel {
        public virtual double Frequency32 { get; protected set; }
        public virtual double Frequency64 { get; protected set; }
        public virtual double Frequency125 { get; protected set; }
        public virtual double Frequency250 { get; protected set; }
        public virtual double Frequency500 { get; protected set; }
        public virtual double Frequency1K { get; protected set; }
        public virtual double Frequency2K { get; protected set; }
        public virtual double Frequency4K { get; protected set; }
        public virtual double Frequency8K { get; protected set; }
        public virtual double Frequency16K { get; protected set; }

        const int MaxValue = 100;

        readonly Random random = new Random();
        readonly DispatcherTimer timer = new DispatcherTimer();

        protected EqualizerViewModel() {
            timer.Interval = TimeSpan.FromMilliseconds(500);
            UpdateData();
        }
        void UpdateData() {
            Frequency32 = random.Next(MaxValue);
            Frequency64 = random.Next(MaxValue);
            Frequency125 = random.Next(MaxValue);
            Frequency250 = random.Next(MaxValue);
            Frequency500 = random.Next(MaxValue);
            Frequency1K = random.Next(MaxValue);
            Frequency2K = random.Next(MaxValue);
            Frequency4K = random.Next(MaxValue);
            Frequency8K = random.Next(MaxValue);
            Frequency16K = random.Next(MaxValue);
        }
        void OnTimedEvent(object source, EventArgs e) {
            UpdateData();
        }

        public void Start() {
            timer.Tick += OnTimedEvent;
            timer.Start();
        }
        public void Stop() {
            timer.Stop();
            timer.Tick -= OnTimedEvent;
        }
    }
}

File: DevExpress-Demo/VB/GaugesDemo.Wpf/ViewModels/EqualizerViewModel.vb
Imports System
Imports System.Windows.Threading

Namespace GaugesDemo

    Public Class EqualizerViewModel

        Public Overridable Property Frequency32 As Double

        Public Overridable Property Frequency64 As Double

        Public Overridable Property Frequency125 As Double

        Public Overridable Property Frequency250 As Double

        Public Overridable Property Frequency500 As Double

        Public Overridable Property Frequency1K As Double

        Public Overridable Property Frequency2K As Double

        Public Overridable Property Frequency4K As Double

        Public Overridable Property Frequency8K As Double

        Public Overridable Property Frequency16K As Double

        Const MaxValue As Integer = 100

        Private ReadOnly random As Random = New Random()

        Private ReadOnly timer As DispatcherTimer = New DispatcherTimer()

        Protected Sub New()
            timer.Interval = TimeSpan.FromMilliseconds(500)
            UpdateData()
        End Sub

        Private Sub UpdateData()
            Frequency32 = random.Next(MaxValue)
            Frequency64 = random.Next(MaxValue)
            Frequency125 = random.Next(MaxValue)
            Frequency250 = random.Next(MaxValue)
            Frequency500 = random.Next(MaxValue)
            Frequency1K = random.Next(MaxValue)
            Frequency2K = random.Next(MaxValue)
            Frequency4K = random.Next(MaxValue)
            Frequency8K = random.Next(MaxValue)
            Frequency16K = random.Next(MaxValue)
        End Sub

        Private Sub OnTimedEvent(ByVal source As Object, ByVal e As EventArgs)
            UpdateData()
        End Sub

        Public Sub Start()
            AddHandler timer.Tick, AddressOf OnTimedEvent
            timer.Start()
        End Sub

        Public Sub [Stop]()
            timer.Stop()
            RemoveHandler timer.Tick, AddressOf OnTimedEvent
        End Sub
    End Class
End Namespace
